### ISSUE TITLE
Move src/KokkosKernels_Macros.hpp to src/common/KokkosKernels_Macros.hpp

### ISSUE THREAD

**Issue by e10harvey** [kokkos/kokkos-kernels#944]

**Comment by lucbv** [comment]
@e10harvey 
I recently merged a small PR to add a macro that checks if the build is host_only, this will help with some tests and code branching.
I put that macro in our configuration file: `kokkos-kernels/cmake/KokkosKernels_config.h.in`, there are a couple of other macros defined there, would it make sense to migrate that content into `KokkosKernels_Macros.hpp`?
If we do so we would probably need to include these files in a lot of places but it might be cleaner than adding macros in the config files?

**Comment by e10harvey** [comment]
Yes. I would argue that `kokkos-kernels/cmake/KokkosKernels_config.h.in` should contain macros populated by CMake and `src/common/KokkosKernels_Macros.hpp` should contain other helper macros that are not touched by cmake. We can move macros from `kokkos-kernels/cmake/KokkosKernels_config.h.in` as part of this issue.

**Comment by lucbv** [comment]
We spent some time on this but it foes not look like it is really worth it, `KokkosKernels_Macros.hpp` was cleaned up and does not include much beyond the config files anymore so we will go ahead and close this now
